### ISSUE TITLE
Simplify and unify environment variable management across all services

### ISSUE THREAD

**Issue by benjaminwfriedman** [benjaminwfriedman/understudy#3]
## Problem Statement

Currently, our platform has inconsistent configuration management:
- Some settings are in `config.py` files, others pulled directly from environment
- No unified approach across services (backend, training, evaluation, model_broker, slm_inference)
- Defaults scattered between code and Kubernetes ConfigMaps
- Direct `os.getenv()` calls throughout codebase
- Sensitive data exposed in docker-compose files

## Requirements

- All settings should have defaults in `config.py` files
- Environment variables should be able to override any setting
- Must work seamlessly in:
  - Kubernetes (local k8s, Azure AKS, AWS EKS, GCP GKE)
  - Docker Compose
  - Local development (outside containers)
- Type-safe, validated configuration
- Consistent approach across all services

#### Step 1.1: Create shared configuration module
- [ ] Create `shared/config_base.py` with base configuration class
- [ ] Implement pydantic_settings BaseSettings pattern
- [ ] Add automatic environment variable override mechanism
- [ ] Add validation and type conversion utilities

#### Step 1.2: Establish naming conventions
- [ ] Document environment variable naming: `UNDERSTUDY_<SERVICE>_<SETTING>`
- [ ] Use double underscore for nested settings
- [ ] Create configuration guidelines in `/docs/configuration.md`

#### Step 2.1: Backend service
- [ ] Enhance `backend/app/core/config.py`
- [ ] Remove all direct `os.getenv()` calls
- [ ] Add missing default values
- [ ] Group settings by category (core, database, providers, training)

#### Step 2.2: Training service
- [ ] Create `training_service/config.py`
- [ ] Migrate from `os.getenv()` to settings object
- [ ] Define defaults for RunPod, Azure, Lambda Cloud settings
- [ ] Add validation for required API keys

#### Step 2.3: Evaluation service
- [ ] Create `evaluation_service/config.py`
- [ ] Define sentence transformer model settings
- [ ] Add Redis and backend URL configurations
- [ ] Set performance tuning defaults

#### Step 2.4: Model broker service
- [ ] Create `model_broker_service/config.py`
- [ ] Define model storage paths with defaults
- [ ] Add streaming and chunk size settings
- [ ] Configure service endpoints

#### Step 2.5: SLM inference service
- [ ] Create `slm_inference_service/config.py`
- [ ] Define model loading settings
- [ ] Add inference parameters defaults
- [ ] Configure batch vs endpoint mode settings

#### Step 3.1: Local development setup
- [ ] Create `.env.example` with all available settings
- [ ] Add `.env` to `.gitignore`
- [ ] Create setup script: `scripts/setup-local-env.sh`
- [ ] Update docker-compose.local.yml to use `.env` file

#### Step 3.2: Remove hardcoded secrets
- [ ] Remove API keys from docker-compose files
- [ ] Move to `.env.example` with placeholder values
- [ ] Document secret management in README

#### Step 3.3: Kubernetes configuration
- [ ] Update ConfigMap with new environment variable names
- [ ] Ensure backward compatibility during migration
- [ ] Update deployment manifests to use ConfigMap consistently
- [ ] Test in local k8s (minikube/kind)

#### Step 4.1: Update import statements
- [ ] Replace `os.getenv()` with `from config import settings`
- [ ] Update all service entry points
- [ ] Update utility modules

#### Step 4.2: Testing
- [ ] Test local development with `.env` file
- [ ] Test Docker Compose deployment
- [ ] Test Kubernetes deployment (local)
- [ ] Test with environment variable overrides

#### Step 5.1: Configuration documentation
- [ ] Document all available settings per service
- [ ] Explain override hierarchy
- [ ] Provide examples for common scenarios

#### Step 5.2: Deployment guides
- [ ] Update deployment documentation
- [ ] Add troubleshooting guide
- [ ] Create migration guide for operators

## Configuration Structure Example

```python
# training_service/config.py
from pydantic_settings import BaseSettings
from typing import Optional

class Settings(BaseSettings):
    # Service identification
    service_name: str = "training-service"
    version: str = "1.0.0"
    
    # Logging
    log_level: str = "INFO"
    log_file: str = "/var/log/training.log"
    
    # Database & Cache
    database_url: str = "postgresql+asyncpg://understudy:password@localhost:5432/understudy"
    redis_url: str = "redis://redis-service:6379"
    
    # Service dependencies
    backend_url: str = "http://backend-service:8000"
    model_broker_url: str = "http://model-broker-service:8003"
    namespace: str = "understudy"
    
    # RunPod configuration
    runpod_api_key: Optional[str] = None
    runpod_gpu_type: str = "NVIDIA GeForce RTX 4090"
    runpod_container_image: str = "runpod/pytorch:2.0.1-py3.10-cuda11.8.0-devel-ubuntu22.04"
    runpod_disk_size: int = 50
    
    # Training defaults
    default_epochs: int = 3
    default_batch_size: int = 4
    default_learning_rate: float = 2e-4
    default_lora_r: int = 16
    default_lora_alpha: int = 32
    
    # HuggingFace
    hf_token: Optional[str] = None
    base_model_path: str = "meta-llama/Llama-3.2-1B"
    
    class Config:
        env_prefix = "UNDERSTUDY_TRAINING_"
        env_file = ".env"
        case_sensitive = False
        
settings = Settings()
```

## Success Criteria

- [ ] All services use consistent configuration approach
- [ ] No direct `os.getenv()` calls in service code
- [ ] All settings have sensible defaults
- [ ] Environment variables can override any setting
- [ ] Works in K8s, Docker, and local development
- [ ] No hardcoded secrets in repository
- [ ] Configuration is type-safe and validated
- [ ] Documentation is complete

## Priority

**High** - This impacts all services and deployment scenarios

## Labels

- enhancement
- architecture
- devops
- configuration

## Estimated Effort

- **Phase 1**: 2-3 hours
- **Phase 2**: 4-5 hours  
- **Phase 3**: 2-3 hours
- **Phase 4**: 3-4 hours
- **Phase 5**: 2 hours

**Total**: ~14-17 hours

## Notes

- Maintain backward compatibility during migration
- Test thoroughly in each deployment environment
- Consider using feature flags for gradual rollout
- May need to coordinate with ongoing deployments
